src/Control/Email/Email.php

```diff
@@ -31,738 +16,646 @@ class Email extends ViewableData
+    }
 
-        $headers = $this->customHeaders;
+    /**
+     * @return string
+     */
+    public function getTemplate()
+    {
+        return $this->template;
+    }
 
-        if ($messageID) {
-            $headers['X-SilverStripeMessageID'] = project() . '.' . $messageID;
+    /**
+     * Set the template to render the email with
+     *
+     * @param string $template
+     * @return $this
+     */
+    public function setTemplate($template)
+    {
+        if (substr($template, -3) == '.ss') {
+            $template = substr($template, 0, -3);
         }
+        $this->template = $template;
 
-        if (project()) {
-            $headers['X-SilverStripeSite'] = project();
-        }
+        return $this;
+    }
 
+    /**
+     * Used by {@link SSViewer} templates to detect if we're rendering an email template rather than a page template
+     *
+     * @return bool
+     */
+    public function IsEmail()
+    {
+        return true;
+    }
 
-        $to = $this->to;
-        $from = $this->from;
-        $subject = $this->subject;
-        if ($sendAllTo = $this->config()->send_all_emails_to) {
-            $subject .= " [addressed to $to";
-            $to = $sendAllTo;
-            if ($this->cc) {
-                $subject .= ", cc to $this->cc";
-            }
-            if ($this->bcc) {
-                $subject .= ", bcc to $this->bcc";
-            }
-            $subject .= ']';
-            unset($headers['Cc']);
-            unset($headers['Bcc']);
-        } else {
-            if ($this->cc) {
-                $headers['Cc'] = $this->cc;
-            }
-            if ($this->bcc) {
-                $headers['Bcc'] = $this->bcc;
-            }
+    /**
+     * @param string $template
+     * @param array|ViewableData $data
+     * @param callable|null $callback
+     * @return static
+     */
+    public static function create_from_callback($template, $data, $callback = null)
+    {
+        // create our email message from the arguments
+        $message = static::create();
+        $message->setTemplate($template);
+        $message->setData($data);
+
+        // if a callback is provided, call it against the message
+        if (is_callable($callback)) {
+            $callback($message);
         }
 
+        return $message;
+    }
 
-        if ($ccAllTo = $this->config()->cc_all_emails_to) {
-            if (!empty($headers['Cc']) && trim($headers['Cc'])) {
-                $headers['Cc'] .= ', ' . $ccAllTo;
-            } else {
-                $headers['Cc'] = $ccAllTo;
-            }
-        }
-
-        if ($bccAllTo = $this->config()->bcc_all_emails_to) {
-            if (!empty($headers['Bcc']) && trim($headers['Bcc'])) {
-                $headers['Bcc'] .= ', ' . $bccAllTo;
-            } else {
-                $headers['Bcc'] = $bccAllTo;
-            }
+    /**
+     * Send the message to the recipients
+     *
+     * @return array|bool true if successful or array of failed recipients
```

Comment:
It might be more concise to have a simple boolean response, and document that the user can call $message->getFailedRecipients() to get the list of failed recipients.